assets/tests/hooks/usePatternsByIdForRoute.test.ts

```diff
@@ -45,75 +49,69 @@ describe("usePatternsByIdForRoute", () => {
-      Api.fetchRoutePatterns as jest.Mock
-    mockFetchRoutePatterns.mockReturnValueOnce(
-      instantPromise([routePatternFactory.buildList(2)])
-    )
-    const { rerender } = renderHook(
-      (id) => {
-        return usePatternsByIdForRoute(id)
-      },
-      { initialProps: "66" }
-    )
-    expect(mockFetchRoutePatterns).toHaveBeenLastCalledWith("66")
+  test("refetches when routeId changes", async () => {
+    const { rerender } = renderHook(usePatternsByIdForRoute, {
+      initialProps: "66",
+    })
+
+    expect(jest.mocked(Api.fetchRoutePatterns)).toHaveBeenLastCalledWith("66")
+
     rerender("39")
-    expect(mockFetchRoutePatterns).toHaveBeenLastCalledWith("39")
+
+    expect(jest.mocked(Api.fetchRoutePatterns)).toHaveBeenLastCalledWith("39")
   })
 
   test("Keeps most recently requested route patterns if previous resolves finishes later", async () => {
-    const patternFor66 = routePatternFactory.build({ routeId: "66" })
-    // const patternsFor66ById = { [patternFor66.id]: patternFor66 }
+    const { promise, resolve } =
+      PromiseWithResolvers<Awaited<ReturnType<typeof Api.fetchRoutePatterns>>>()
+
     const patternFor39 = routePatternFactory.build({ routeId: "39" })
     const patternsFor39ById = { [patternFor39.id]: patternFor39 }
 
-    let patternsFor66DidResolve = false
+    const routeId1 = routeFactory.build()
 
-    const mockFetchRoutePatterns: jest.Mock =
-      Api.fetchRoutePatterns as jest.Mock
+    jest.mocked(Api.fetchRoutePatterns).mockReturnValue(promise)
 
-    mockFetchRoutePatterns
-      .mockReturnValueOnce(
-        new Promise((res) =>
-          setTimeout(() => {
-            patternsFor66DidResolve = true
-            return res([patternFor66])
-          }, 250)
-        )
-      )
-      .mockReturnValueOnce(instantPromise([patternFor39]))
-    const { rerender, result } = renderHook(
-      (id) => {
-        return usePatternsByIdForRoute(id)
-      },
-      { initialProps: "66" }
+    const { rerender, result } = renderHook(usePatternsByIdForRoute, {
+      initialProps: routeId1.id,
+    })
+
+    expect(jest.mocked(Api.fetchRoutePatterns)).toHaveBeenLastCalledWith(
+      routeId1.id
     )
-    expect(mockFetchRoutePatterns).toHaveBeenLastCalledWith("66")
+
     expect(result.current).toBeNull()
+
+    jest.mocked(Api.fetchRoutePatterns).mockResolvedValue([patternFor39])
+
     rerender("39")
-    expect(mockFetchRoutePatterns).toHaveBeenLastCalledWith("39")
-    expect(result.current).toEqual(patternsFor39ById)
 
-    expect(patternsFor66DidResolve).toBe(false)
+    expect(jest.mocked(Api.fetchRoutePatterns)).toHaveBeenLastCalledWith("39")
+    await waitFor(() => expect(result.current).toEqual(patternsFor39ById))
+
+    act(() =>
+      resolve(routePatternFactory.buildList(2, { routeId: routeId1.id }))
+    )
 
-    await waitFor(() => expect(patternsFor66DidResolve).toBe(true))
+    await waitFor(() => expect(result.current).toEqual(patternsFor39ById))
```

Comment:
If this test finishes executing quickly enough that the promise doesn't have time to resolve (or the hook doesn't have time to respond to the promise resolving), would this still pass?

Also, is it worth having this assertion here followed by the same assertion just below?

Reply:
[I think we can drop the assertion after this](https://github.com/mbta/skate/compare/1353e383a98b19ba4bd9ffb65da5fe0a9aa7f18d..2bd018703ea29ec1e59a9d6f44857d3dc8c828ad)